core/newlib_syscalls.c

```diff
@@ -41,12 +42,8 @@ IRAM caddr_t _sbrk_r (struct _reent *r, int incr)
 }
 
 /* syscall implementation for stdio write to UART */
-__attribute__((weak)) long _write_r(struct _reent *r, int fd, const char *ptr, int len )
+__attribute__((weak)) long _write_stdout_r(struct _reent *r, int fd, const char *ptr, int len )
```

Comment:
Do we really need it to be `weak`? We already have weak `_write_r` in case user needs to override it completely. And have a run-time selection of stdout write function.
So, I think it might be a static function without underscore prefix.